packages/Toggle/src/Toggle.js

```diff
@@ -1,7 +1,357 @@
+import { createDataProp, emotion } from '@leafygreen-ui/lib';
+import { colors } from '@leafygreen-ui/theme';
+
+const { css, cx } = emotion;
+
+const toggleInput = createDataProp('toggle-input');
+const toggleGroove = createDataProp('toggle-groove');
+
+const transitionInMS = 150;
+
+const inputStyle = css`
+  margin: 0;
+  position: absolute;
+  left: 100%;
+  top: 100%;
+  pointer-events: none;
+  opacity: 0;
+`;
+
+// We use a div for the focus state rather than a pseudo-element
+// because said pseudo-element would need to be on the label element
+// which can't use the contained input's focus pseudo-class.
+const focusStateStyle = css`
+  transition: all ${transitionInMS}ms ease-in-out;
+  position: absolute;
+  top: -2px;
+  bottom: -2px;
+  left: -2px;
+  right: -2px;
+  border: 2px solid rgba(67, 177, 229, 0.25);
+  border-radius: 50px;
+  opacity: 0;
+  transform: scale(0.8);
+
+  ${toggleInput.selector}:focus ~ & {
+    opacity: 1;
+    transform: scale(1);
+  }
+`;
+
+const getContainerStyles = ({ size, disabled }) => {
+  const sizeStyle = {
+    default: css`
+      height: 32px;
+      width: 62px;
+    `,
+
+    small: css`
+      height: 22px;
+      width: 40px;
+    `,
+
+    xsmall: css`
+      height: 14px;
+      width: 26px;
+    `,
+  };
+
+  return cx(
+    css`
+      position: relative;
+      display: inline-block;
+      cursor: ${disabled ? 'not-allowed' : 'pointer'};
+    `,
+    sizeStyle[size],
+  );
+};
+
+const getGrooveStyles = ({ variant, checked, disabled }) => {
+  let baseStyle = `
+    transition: ${transitionInMS}ms all ease-in-out, 0 background-color linear;
+    display: inline-block;
+    overflow: hidden;
+    flex-shrink: 0;
+    position: relative;
+    border-radius: 50px;
+    overflow: hidden;
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    border: 1px solid;
+
+    // We're animating this pseudo-element in order to give the toggle groove
+    // background an animation in and out.
+    &:before {
+      content: '';
+      transition: ${transitionInMS}ms all ease-in-out;
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      left: 0;
+      right: 0;
+      border-radius: 50px;
+      background-color: ${colors.mongodb.blue};
+      box-shadow: inset 0 0 10px rgba(0, 0, 0, 0.1);
+      opacity: 0;
+      transform: scale(0.85);
+    }
+  `;
+
+  if (disabled) {
+    baseStyle += `
+      &:before {
+        opacity: 0;
+      }
+    `;
+  } else if (checked) {
+    baseStyle += `
+      // We set background-color here to avoid a small issue with overflow clipping
+      // that makes this look less seamless than it should.
+      background-color: ${colors.mongodb.blue};
+      border-color: #2e9ed3;
+      transition-delay: ${transitionInMS}ms;
+
+      &:before {
+        transform: scale(1);
+        opacity: 1;
+      }
+    `;
+  }
+
+  const variants = {
+    default: (() => {
+      let variantStyle = '';
+
+      if (disabled) {
```

Comment:
You can just directly return css() with the styles in there, can also be first line in the file.

Reply:
Ahh, good point. Was converting stuff to practice happy-path better, and that would take it further in that direction. Love it.

Reply:
Done and it feels good.